### ISSUE TITLE
[Bug] saved_query rule type not supported by detection-rules CLI

### ISSUE THREAD

**Issue by Mikaayenson** [elastic/detection-rules#3192]
**Describe the bug**
When trying to import rules `saved_query` rule types or create rules using the `create-rule` CLI command, the logic crashes.  The import may be resolved once https://github.com/elastic/detection-rules/issues/3191 is complete, but we should double check the create command supports the feature.

**To Reproduce**
1. export a saved_query rule 
2. Try to import with the CLI
3. See error

**Expected behavior**
Both importing saved_query rule types and using the create-rule command should not crash.

**Screenshots**

![image (19)](https://github.com/elastic/detection-rules/assets/1636709/f357ee7c-3785-4f95-b781-a1d9f34bf600)

**Additional context**
- Reported by a community member: [link](https://elasticstack.slack.com/archives/C016E72DWDS/p1697466234551809?thread_ts=1697367379.109409&cid=C016E72DWDS)

**Comment by terrancedejesus** [comment]
@Mikaayenson 

It doesn't look like we support saved query rule types any more after some review.
Checking the [detection engine ](https://github.com/elastic/kibana/tree/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types)code in Kibana shows that the rule type does not exist, nor is it listed in the detection engine [index](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/index.ts).

Look at a fairly new stack version, 8.10.3, it doesn't appear that the saved query rule type is available either.
<img width="1245" alt="Screenshot 2023-10-17 at 11 01 07 AM" src="https://github.com/elastic/detection-rules/assets/99630311/1dbecd03-2a14-40fb-ad62-c1e742ccc0c7">

**Comment by Mikaayenson** [comment]
Makes sense, since the logic was removed in a [prior PR](https://github.com/elastic/detection-rules/pull/1084/files#diff-f0ee953fa3e8e36f9bb78ec7c96b576a45af39b6d648373fd7a804c243d49446L114-L117). So we will remove any occurrences of this from our CLI. 

cc. @zeuhl

**Comment by Mikaayenson** [comment]
@banderror Do you have references in kibana to `saved_query`. We want to check if this query type exists before we remove the last references. We used to have schema entries a long time ago that looked like:

```
   with jsl.Scope(SAVED_QUERY) as saved_id_scope:
         saved_id_scope.index = jsl.ArrayField(jsl.StringField(), required=False)
         saved_id_scope.saved_id = jsl.StringField(required=True)
         saved_id_scope.type = jsl.StringField(enum=[SAVED_QUERY], required=True, default=SAVED_QUERY)
```

Did `saved_query` perhaps get renamed to something else?

**Comment by zeuhl** [comment]
Hello !!!

The saved query already exist in fact, this rule type appear when you select a "saved query" in "Custom query" rule type, the option is : "Load saved query "600003" dynamically on each rule execution" : 
![image](https://github.com/elastic/detection-rules/assets/6780173/7891d63a-6b4f-472b-8339-0267e1572b77)
![image](https://github.com/elastic/detection-rules/assets/6780173/0672c3e6-cede-42c5-ab82-269c826704d1)
![image](https://github.com/elastic/detection-rules/assets/6780173/5fb894ee-af52-4bff-946a-e099eccbddaf)

This feature load the query on each execution. (it's a hard link between saved_query object and rule object in Kibana)

**Comment by banderror** [comment]
@zeuhl Thank you for this super-detailed explanation, this is correct. I can only add that what you described is an internal implementation detail of this page, and Custom Query rules at the schema level now also support linking to a saved query instead of specifying a normal, inline query in the rule itself.

@Mikaayenson I think the TL;DR here would be:

- The `saved_query` rule type still exists in the code. We support it in the Detections API, and rules of this type are executed by the Detection Engine. We haven't removed any code related to `saved_query` rules.
- If you need to create a rule via the API that should be linked to a saved query, you can still use the `saved_query` rule type.
- If you need to update an existing Custom Query rule (of type `query`) so that it will point to a saved query instead of having its own query, you can now do it. I think in the past, you'd have to create a new rule.

We don't link to any saved queries from prebuilt rules and don't have any plans for doing it, right? If so, it should be safe to remove any code for this rule type from the https://github.com/elastic/detection-rules schemas/tooling.

**Comment by Mikaayenson** [comment]
Thanks @banderror for confirming. @zeuhl We currently do not use `saved_query` within our rule management and do not plan to link prebuilt rules to saved queries. We're planning to remove the remaining mentions for now. 

Based on your initial slack question you mentioned exploring the code while debugging the missing alert suppression fields and looked for the saved_query schemas. Is there a specific use case that you would like supported regarding `saved_query`  (e.g. `saved_id` field added to our rule schema to reference `saved_query` objects) ?  If so, we should probably migrate this to a new FR issue to help you with your internal workflows!

**Comment by zeuhl** [comment]
Thank you very much @Mikaayenson for this information, and I apologize for the delay in my response.

In fact, I've been using Elasticsearch Security from the beginning, and at that time, we linked the rules with saved queries to facilitate analysis and whitelisting from the Kibana Discover view.

Today, things have evolved significantly, with the introduction of exceptions and the alert view changing things substantially. The initial goal was to use the code in this repo to translate our rules into the format and take advantage of schema validation and other libraires from this repo.

However, I have moved my rules to Query type and transferred all my filters (painless, lucene) to exceptions . All my use cases (with some adjustments in the code to use my tags, internal naming convention, link exceptions, ...) are now in TOML format, allowing me to use this repo without constraints.

Thanks again for your responsiveness.
